geonotebook/vis/geoserver/geoserver.py

```diff
@@ -75,6 +76,52 @@ def coverages(self, store):
         else:
             return {}
 
+    @staticmethod
+    def generate_colormap(colormap, minimum, maximum):
+        """ colormap can be a list of dicts
+        or a matplotlib colormap. Either case
+        the returned object will be a list of dicts.
+        """
+
+        def range_count(start, stop, count):
+            """ Generates a list with given start
+            stop and count with linear spacing
+            e.g. range_count(1, 3, 5) = [1., 1.5, 2., 2.5, 3.]
+            """
+
+            step = (stop - start) / float(count-1)
+            return [start + i * step for i in xrange(count)]
+
+        def rgba2hex(rgba):
+            """ Converts rgba values to hex """
+
+            # Slice the tuple so that
+            # we don't get alpha and
+            # convert values to 8 bit ints
+            rgb = tuple([int(255 * i) for i in rgba[:3]])
+            hex = "#{}".format(struct.pack('BBB',*rgb).encode('hex'))
```

Comment:
small nitpick - ```hex``` is a built in function -  also  lets just ```return "#{}".format(....)``` rather than assigning and returning.